### PR TITLE
[10.0] [FIX] Solve encoding issue and add AMEX support

### PR DESCRIPTION
**Pull request by sebastienbeau**
This PR solve 3 issues:

- importing AMEX qif bank statement (header is CCARD)
- solving encoding issue by guessing it with the lib "chardet" (optionnal lib)
- make inheritable the way to parse the date

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Changing an external dependency in a stable version is not convenient, as it makes all existing installations to break.

**Comment by sebastienbeau** [comment]
@pedrobaeza you are right, I can make it optionnal.

**Comment by sebastienbeau** [comment]
I have changed my code, now I use utf-8 by default (this should work with the majority of case) if you had the chardet lib, it will try to guess the right encoding.
Also I have refactored the code to be able to inherit the way to parse the date in custom module

File: account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py [hunk 1]
```diff
@@ -12,6 +12,14 @@
 from odoo import api, models
 from odoo.exceptions import UserError
 
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    import chardet
+except (ImportError, IOError) as err:
+    _logger.debug(err)
```
Comment (pedrobaeza) [inline account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py]: Put here `chardet = False` instead for easy handling

File: account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py [hunk 2]
```diff
@@ -20,10 +28,26 @@ class AccountBankStatementImport(models.TransientModel):
     def _check_qif(self, data_file):
         return data_file.strip().startswith('!Type:')
 
+    def _get_qif_encoding(self, data_file):
+        try:
```
Comment (pedrobaeza) [inline account_bank_statement_import_qif/wizards/account_bank_statement_import_qif.py]: Don't put the message each time. With the other change, you can do:
```
if chardet:
    chardet.detect(data_file)['encoding']
else:
    return 'utf-8'
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And please change module version number

**Comment by sebastienbeau** [comment]
Thanks for your feedback Pedro !
I have done the fixes

**Review by pedrobaeza — APPROVED** [review]

**Review by bealdav — APPROVED** [review]

**Pull request merged**